mappings/net/minecraft/block/RedstoneWireBlock.mapping

```diff
@@ -19,7 +19,7 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
 		ARG 1 view
 		ARG 2 pos
 		ARG 3 dir
-	METHOD method_10478 (Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_2350;)Z
+	METHOD method_10478 couldConnectTo (Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_2350;)Z
```

Comment:
Is there already a `canConnectTo`?

Reply:
The method returns true if the wire could connect to a non-wire block (nothing's in the way, either a full block or a redstone component wire is there, etc.). The wire will only actually connect to the block if it could connect to it but not the blocks on the side:

```java
    @Override
    public int getWeakRedstonePower(BlockState state, BlockView view, BlockPos pos, Direction facing) {
        if (!wiresGivePower) {
            return 0;
        }
        
        int power = state.get(RedstoneWireBlock.POWER);
        if (power == 0) {
            return 0;
        }
        
        if (facing == Direction.UP) {
            return power;
        }
        
        EnumSet<Direction> directions = EnumSet.noneOf(Direction.class);
        
        for (Direction direction : Direction.Type.HORIZONTAL) {
            if (couldConnectTo(view, pos, direction)) {
                directions.add(direction);
            }
        }

        if (facing.getAxis().isHorizontal() && directions.isEmpty()) {
            return power;
        }

        if (directions.contains(facing) && !directions.contains(facing.rotateYCounterclockwise()) && !directions.contains(facing.rotateYClockwise())) {
            return power;
        }
        
        return 0;
    }
```

Reply:
`Direction.Type.HORIZONTAL`
Direction.Type? What happened to `Direction.Axis`?

Reply:
It still exists. Direction.Axis is X, Y, Z, and Direction.Type is HORIZONTAL, VERTICAL.

Reply:
Ooh, okay. Didn't know they had that.